zhaquirks/sonoff/swv.py

```diff
@@ -0,0 +1,112 @@
+"""Sonoff SWV - Zigbee smart water valve."""
+
+from zigpy.profiles import zha
+from zigpy.quirks import CustomCluster, CustomDevice
+import zigpy.types as t
+from zigpy.zcl.clusters.general import (
+    Basic,
+    Identify,
+    OnOff,
+    Ota,
+    PollControl,
+    PowerConfiguration,
+    Time,
+)
+from zigpy.zcl.clusters.homeautomation import Diagnostic
+from zigpy.zcl.clusters.measurement import FlowMeasurement
+
+from zhaquirks.const import (
+    DEVICE_TYPE,
+    ENDPOINTS,
+    INPUT_CLUSTERS,
+    MODELS_INFO,
+    OUTPUT_CLUSTERS,
+    PROFILE_ID,
+)
+
+SONOFF_CLUSTER_FC11_ID = 0xFC11
+SONOFF_CLUSTER_FC57_ID = 0xFC57
+ATTR_SONOFF_VALVE_STATUS = 0x500C
+
+
+class ValveStatusBitmap(t.bitmap8):
+    """Valve status value enum."""
+
+    Water_Shortage = 0x1
+    Water_Leakage = 0x2
+
```

Comment:
I think you want Normal (0) and probably Leakage and Shortage (3).

Reply:
So the absence of Normal (0) and Leakage and Shortage (3) was intentional - if you look at class you will see that this is a bitfield not an enum. Leakage and Shortage (3) would just be Water_Shortage (1) & Water_Leakage (2)

If you look what I've done in ZHR PR (https://github.com/zigpy/zha/pull/189), I'm actually exporting two binary sensors, one for each bit. I personally think this is a lot nicer than how they did it in Zigbee2MQTT.

![Screenshot 2024-08-31 201932](https://github.com/user-attachments/assets/41a39e89-a8f2-4ec0-b68d-71f82ef4a335)

Reply:
Ah, I see, I missed the bitmap8.  
Two related comments: 

- The comment says enum :)
- While this is fine for the sensor, the other part is the response when reading the attribute directly via Manage Zigbee Device. This is not a big issue, but it'd be nice to have the full range of values IMO.

Reply:
Actually, is there any reason you cannot use an enum here and check for the specific values on the other PR?